coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/SqlNodeConverter.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2022-2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.rel2trino;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.calcite.sql.SqlBasicTypeNameSpec;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlDataTypeSpec;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.SqlSelect;
+import org.apache.calcite.sql.SqlTypeNameSpec;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.sql.util.SqlShuttle;
+import org.apache.calcite.sql.validate.SqlValidatorUtil;
+
+import static org.apache.calcite.rel.rel2sql.SqlImplementor.*;
+import static org.apache.calcite.sql.parser.SqlParserPos.*;
+
+
+/**
+ * SqlNodeConverter transforms the sqlNodes
+ * in the input SqlNode representation to be compatible with Trino engine.
+ * The transformation may involve change in operator, reordering the operands
+ * or even re-constructing the SqlNode.
+ *
+ * NOTE: This is a temporary class which hosts certain transformations which were previously done in RelToTrinoConverter.
+ * This class will be refactored once standardized CoralIR is integrated in the CoralRelNode to trino SQL translation path.
+ */
+public class SqlNodeConverter extends SqlShuttle {
+  public SqlNodeConverter() {
+  }
+
+  @Override
+  public SqlNode visit(final SqlCall call) {
+    SqlCall transformedSqlCall = getTransformedSqlCall(call);
+    return super.visit(transformedSqlCall);
+  }
+
+  private static SqlCall getTransformedSqlCall(SqlCall sqlCall) {
+    switch (sqlCall.getOperator().kind) {
+      case EQUALS:
+      case GREATER_THAN:
+      case GREATER_THAN_OR_EQUAL:
+      case LESS_THAN:
+      case LESS_THAN_OR_EQUAL:
+      case NOT_EQUALS:
+        return castOperandsToVarchar(sqlCall);
+      case SELECT:
+        return getTransformedSqlSelectSqlCall(sqlCall);
+      default:
+        return sqlCall;
+    }
+  }
+
+  /**
+   * Coral IR generates a SqlNode representation where operands of a relational operator may not be compatible.
```

Comment:
Coral IR allows implicit casting to VARCHAR from other data types such as INTs, etc. However, Trino requires explicit casting.